tools/roslaunch/src/roslaunch/config.py

```diff
@@ -445,10 +448,11 @@ def load_config_default(roslaunch_files, port, roslaunch_strs=None, loader=None,
     load_roscore(loader, config, verbose=verbose)
 
     # load the roslaunch_files into the config
-    for f in roslaunch_files:
+    for i, f in enumerate(roslaunch_files):
+        args = roslaunch_args[i] if roslaunch_args else None
```

Comment:
Currently this requires `roslaunch_args` to be either `None` or at least have the same length as `roslaunch_files`. It would be good to support the usage where `roslaunch_args` contains less items than `roslaunch_files` since a user might only pass fewer `roslaunch_args` since the other launch files don't need arguments.

Reply:
I agree that the interface is not very pretty or ideal, but I'm having trouble seeing any way to determine which roslaunch file should get an arg list apart from 2 equal-length lists. Otherwise, if the user gives multiple launch files, but only one arg list, how do we decide which file to give the list to? Can we just document that any launch files w/o args must pass in a `None`?

Reply:
Or perhaps we pass that args in as a dictionary with roslaunch filenames as keys and arglists as values? That would remove all ambiguity... And `dict.get` already plays nice by defaulting to `None` for nonexistent keys. I'll work on that tack and see how the interface feels.

Reply:
How about just allowing the `roslaunch_args` list to be shorter. That avoids that the user needs to append `None` for launch files at the end which don't have arguments.

Reply:
That's an option. What are your thoughts on my later commits using a dictionary? Seems to give the best of both worlds to me, but maybe I've got blinders on. I can revert those commits and go with this instead.

Reply:
Having two lists which are correlated by their index is certainly not very elegant.

But with the current dict I do see two downsides:

* the repetition of the launch file
* the changed semantic when using arguments for launch files which appear more than once

  * Before calling it with `[foo, foo, bar]` was straight forward. But when passing arguments those need to be grouped by the file name `{foo: [args-for-first-foo, args-for-second-foo], bar: [args-for-bar]}` which seems rather unintuitive.

Another alternative might be to pass the filenames and their arguments in a single parameter. Each item in the list could be either a plain filename or a tuple where the first element is the filename and the second element contains the arguments. That might provide a nicer API while still being backward compatible?

Reply:
I like your last idea. It's intuitive and should be easy to implement as well as understand for users. I'm on it.